primitives/account/src/logs.rs

```diff
@@ -0,0 +1,364 @@
+            } => validator_address == address,
+            Log::Slash {
+                validator_address, ..
+            } => validator_address == address,
+            Log::RevertContract { contract_address } => contract_address == address,
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+#[cfg_attr(feature = "serde-derive", derive(serde::Serialize, serde::Deserialize))]
+pub struct TransactionLog {
+    #[serde(rename = "hash")]
+    pub tx_hash: Blake2bHash,
+    pub logs: Vec<Log>,
+}
+
+impl TransactionLog {
+    pub fn new(tx_hash: Blake2bHash, logs: Vec<Log>) -> Self {
+        Self { tx_hash, logs }
+    }
+}
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+#[cfg_attr(feature = "serde-derive", derive(serde::Serialize, serde::Deserialize))]
+// Renaming affects only the struct names and thus their tag, the "type" field.
+#[serde(rename_all = "kebab-case", tag = "type")]
+pub enum BlockLog {
+    #[serde(rename_all = "camelCase")]
+    AppliedBlock {
+        #[serde(rename = "inherents")]
+        inherent_logs: Vec<Log>,
+        block_hash: Blake2bHash,
+        block_number: u32,
+        timestamp: u64,
+        #[serde(rename = "transactions")]
+        tx_logs: Vec<TransactionLog>,
+    },
+
+    #[serde(rename_all = "camelCase")]
+    RevertedBlock {
+        #[serde(rename = "inherents")]
+        inherent_logs: Vec<Log>,
+        block_hash: Blake2bHash,
+        block_number: u32,
+        #[serde(rename = "transactions")]
+        tx_logs: Vec<TransactionLog>,
+    },
+}
+
+impl BlockLog {
+    pub fn is_revert_block_log(&self) -> bool {
+        match self {
+            BlockLog::AppliedBlock { .. } => false,
+            BlockLog::RevertedBlock { .. } => true,
+        }
+    }
+}
+
+// Account Info is used as a return type of transactions to the blockchain and stores the logs and receipt associated with the transaction
+#[derive(Debug, PartialEq, Eq)]
+pub struct AccountInfo {
+    pub receipt: Option<Vec<u8>>,
+    pub logs: Vec<Log>,
+}
+
+impl AccountInfo {
+    pub fn new(receipt: Option<Vec<u8>>, logs: Vec<Log>) -> Self {
+        Self { receipt, logs }
+    }
+
+    pub fn with_option_receipt<T: BeSerialize>(op_info: OperationInfo<Option<T>>) -> Self {
+        let serialized_receipt = match op_info.receipt {
+            Some(receipt) => Some(receipt.serialize_to_vec()),
+            None => None,
+        };
+        AccountInfo::new(serialized_receipt, op_info.logs)
+    }
+}
+
+impl<T: BeSerialize> From<OperationInfo<T>> for AccountInfo {
+    fn from(op_info: OperationInfo<T>) -> Self {
+        AccountInfo::new(
+            op_info.receipt.map(|receipt| receipt.serialize_to_vec()),
+            op_info.logs,
+        )
+    }
+}
+
+// Batch Info is used as a return type of multiple transactions or batches applied to the blockchain.
+// It stores the transaction logs, inherent logs and receipts associated with the multiple transactions of the batch.
+#[derive(Default, Eq, PartialEq, Debug)]
```

Comment:
Same here, aren't we missing `#[serde(rename_all = "camelCase")]`?

Reply:
AFAIK this struct is never sent as JSON.